elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -979,8 +973,8 @@ protected Object getValue(String fieldName) {
      * @param toAdd the to add
      */
     protected void addToCollection(Collection collection, String collectionName, PersistentResource toAdd) {
-        checkFieldAwarePermissions(UpdatePermission.class, collectionName);
-        toAdd.checkFieldAwarePermissions(ReadPermission.class, collectionName);
+        checkFieldAwarePermissions(UpdatePermission.class, collectionName, toAdd, null);
```

Comment:
The permission should be looking at collections (not null) for the changespec.  This is a general comment for wherever we should be passing a collection as the change spec